ccl/changefeedccl: TestChangefeedMultiPodTenantPlanning failed

ccl/changefeedccl.TestChangefeedMultiPodTenantPlanning [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/19698420?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/19698420?buildTab=artifacts#/) on release-25.2 @ [c93a49bc2530662e673d8c7b2d42d19f2f47081e](https://github.com/cockroachdb/cockroach/commits/c93a49bc2530662e673d8c7b2d42d19f2f47081e):


```
=== RUN   TestChangefeedMultiPodTenantPlanning
    test_log_scope.go:165: test logs captured to: /artifacts/tmp/_tmp/a152b17b8fdc5b459db71a1944ab6ad5/logTestChangefeedMultiPodTenantPlanning146057794
    test_log_scope.go:76: use -show-logs to present logs inline
    helpers_test.go:1119: making pulsar feed factory
    changefeed_test.go:10105: 
        	Error Trace:	pkg/ccl/changefeedccl/changefeed_test.go:10105
        	Error:      	Not equal: 
        	            	expected: 2
        	            	actual  : 1
        	Test:       	TestChangefeedMultiPodTenantPlanning
    panic.go:635: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/a152b17b8fdc5b459db71a1944ab6ad5/logTestChangefeedMultiPodTenantPlanning146057794
--- FAIL: TestChangefeedMultiPodTenantPlanning (14.64s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/cdc
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedMultiPodTenantPlanning.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-50717

--- a/pkg/ccl/changefeedccl/changefeed_test.go
+++ b/pkg/ccl/changefeedccl/changefeed_test.go
@@ -10809,6 +10809,23 @@ func TestChangefeedMultiPodTenantPlanning(t *testing.T) {
 	sql1.Exec(t, "CREATE TABLE bar (b INT PRIMARY KEY)")
 	sql1.Exec(t, "INSERT INTO bar VALUES (1), (2)")
 
+	// Force foo and bar into separate ranges. Without this, the two tiny
+	// tables may share a single range, in which case PartitionSpans returns
+	// one partition and the two-aggregator assertion below flakes.
+	sql1.Exec(t, "ALTER TABLE foo SPLIT AT VALUES (0)")
+	sql1.Exec(t, "ALTER TABLE bar SPLIT AT VALUES (0)")
+	testutils.SucceedsSoon(t, func() error {
+		var fooRangeID, barRangeID int
+		sql1.QueryRow(t,
+			"SELECT range_id FROM [SHOW RANGES FROM TABLE foo] LIMIT 1").Scan(&fooRangeID)
+		sql1.QueryRow(t,
+			"SELECT range_id FROM [SHOW RANGES FROM TABLE bar] LIMIT 1").Scan(&barRangeID)
+		if fooRangeID == barRangeID {
+			return errors.Newf("foo and bar still share range %d", fooRangeID)
+		}
+		return nil
+	})
+
 	foo := feed(t, feedFactory, "CREATE CHANGEFEED FOR foo, bar")
 	defer closeFeed(t, foo)
 

